pkg/datasource/opensergo/opensergo.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 1999-2020 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opensergo
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/alibaba/sentinel-golang/ext/datasource"
+	"github.com/alibaba/sentinel-golang/logging"
+	"github.com/alibaba/sentinel-golang/util"
+	"github.com/opensergo/opensergo-go/pkg/api"
+	"github.com/opensergo/opensergo-go/pkg/client"
+	"github.com/opensergo/opensergo-go/pkg/configkind"
+	"github.com/opensergo/opensergo-go/pkg/model"
+	"github.com/pkg/errors"
+)
+
+type OpenSergoDataSource struct {
```

Comment:
We might try to *fully* implement `DataSource` interface.

Reply:
done.